### PR TITLE
Add more wake lock tests

### PR DESCRIPTION
**Pull request by Honry**
- test in an insecure context
  - test for invoking 'cancel()' twice
  - test when document is hidden
  - test for applicability when screen is switched off

https://github.com/w3c/wake-lock/issues/111

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/20605)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/302262055)

File: wake-lock/wakelock-applicability-manual.https.html
Comment (marcoscaceres) [inline wake-lock/wakelock-applicability-manual.https.html]: This is unclear, are you asking the tested to wait 60 seconds?

File: wake-lock/wakelock-applicability-manual.https.html
Comment (marcoscaceres) [inline wake-lock/wakelock-applicability-manual.https.html]: nit: unquoted attributes.

File: wake-lock/wakelock-applicability-manual.https.html
Comment (marcoscaceres) [inline wake-lock/wakelock-applicability-manual.https.html]: Nit: "applicable" is a somewhat confusing term. Maybe "usable"? dunno.

File: wake-lock/wakelock-applicability-manual.https.html
Comment (marcoscaceres) [reply wake-lock/wakelock-applicability-manual.https.html]: I theory, the tests will run by the time the tested turns the screen off and then on again. So likely they only have to wait maybe 1 or 2 seconds, right?

File: wake-lock/wakelock-cancel-twice.https.html
Comment (marcoscaceres) [inline wake-lock/wakelock-cancel-twice.https.html]: Cute hack :) However, if we end up adding more tests to this file, people looking at the test might get annoyed if they have to scroll to the top to find out what the test does. Just inline the testable assertion here and maybe generalize the doc's title.

File: wake-lock/wakelock-insecure-context.html
Comment (marcoscaceres) [inline wake-lock/wakelock-insecure-context.html]: as above.

**Review by marcoscaceres — APPROVED** [review]
Couple of nits, but these are looking great @Honry! 🏆 

I'd like to get a review from @andrey-logvinov before we merge. @andrey-logvinov, can you please double check that the algorithms are being fully tested?

File: wake-lock/wakelock-applicability-manual.https.html
Comment (Honry) [reply wake-lock/wakelock-applicability-manual.https.html]: I didn't mean waiting, I just wanted to make sure the interaction should be complete within 60 seconds, before test timeout. Anyway, I will improve the description.

File: wake-lock/wakelock-applicability-manual.https.html
Comment (Honry) [reply wake-lock/wakelock-applicability-manual.https.html]: Hmm.. This sentence is referenced from [spec](https://w3c.github.io/wake-lock/#dfn-the-wake-lock-is-applicable), I think we can leave it as is.

> The screen wake lock MUST NOT be applicable after the screen is manually swiched off by the user until it is switched on again. Manually switching off the screen MUST NOT affect the applicability of the system wake lock.

File: wake-lock/wakelock-cancel-twice.https.html
Comment (Honry) [reply wake-lock/wakelock-cancel-twice.https.html]: :) Will fix.

File: wake-lock/wakelock-applicability-manual.https.html
Comment (marcoscaceres) [reply wake-lock/wakelock-applicability-manual.https.html]: ah, thanks for clarifying... ok, yeah, it makes sense in the context of the spec.

**Comment by Honry** [comment]
> I'd like to get a review from @andrey-logvinov before we merge. @andrey-logvinov, can you please double check that the algorithms are being fully tested?

@andrey-logvinov, friendly ping. :)

**Comment by marcoscaceres** [comment]
Merging, as it's been quite a few days - @andrey-logvinov, could you please still have a look at this and file any bugs as appropriate?

**Pull request merged**
